agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFile.java

```diff
@@ -140,7 +140,12 @@ private StatusFile() {}
 
   // visible for testing
   static boolean shouldWrite() {
-    return DiagnosticsHelper.useAppSvcRpIntegrationLogging();
+    return DiagnosticsHelper.useAppSvcRpIntegrationLogging() && !isReadOnlyFileSystem();
```

Comment:
Let's init `shouldWrite` value in static initializer, then we can log once something like:

      startupLogger.info(
          "Detected running on a read-only file system, app service status json fiule won't be written. If this is unexpected, please check that the process has write access to the directory: "
              + dir.getAbsolutePath());